domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt

```diff
@@ -1278,6 +1279,126 @@ class ProfileManagementControllerTest {
+    assertThat(profile.profileType).isEqualTo(ProfileType.SOLE_LEARNER)
+    assertThat(profile.isAdmin).isEqualTo(true)
+    assertThat(profile.avatar.avatarImageUri).isEqualTo("")
+    assertThat(profile.avatar.avatarColorRgb).isEqualTo(-1)
+  }
+
+  @Test
+  fun testUpdateProfile_updateMultipleFields_invalidName_checkUpdateFailed() {
+    setUpTestApplicationComponent()
+    profileTestHelper.createDefaultAdminProfile()
+
+    val updateProvider = profileManagementController.updateNewProfileDetails(
+      PROFILE_ID_0,
+      ProfileType.SOLE_LEARNER,
+      null,
+      -1,
+      "John123",
+      isAdmin = true
+    )
+    val failure = monitorFactory.waitForNextFailureResult(updateProvider)
+
+    assertThat(failure).hasMessageThat().contains("John123 does not contain only letters")
+  }
+
+  @Test
+  fun testUpdateProfile_updateMultipleFields_nullAvatarUri_setsAvatarColorSuccessfully() {
+    setUpTestApplicationComponent()
+    profileTestHelper.createDefaultAdminProfile()
+
+    val updateProvider = profileManagementController.updateNewProfileDetails(
+      PROFILE_ID_0,
+      ProfileType.SOLE_LEARNER,
+      null,
+      -11235672,
+      "John",
+      isAdmin = true
+    )
+    monitorFactory.waitForNextSuccessfulResult(updateProvider)
+
+    val profileProvider = profileManagementController.getProfile(PROFILE_ID_0)
+    val profile = monitorFactory.waitForNextSuccessfulResult(profileProvider)
+
+    assertThat(profile.avatar.avatarImageUri).isEqualTo("")
+    assertThat(profile.avatar.avatarColorRgb).isEqualTo(-11235672)
+  }
+
+  @Test
+  fun testUpdateProfile_updateMultipleFields_invalidProfileId_checkUpdateFailed() {
+    setUpTestApplicationComponent()
+    profileTestHelper.createDefaultAdminProfile()
+
+    val updateProvider = profileManagementController.updateNewProfileDetails(
+      PROFILE_ID_3,
+      ProfileType.SOLE_LEARNER,
+      null,
+      -1,
+      "John",
+      isAdmin = true
+    )
+    val failure = monitorFactory.waitForNextFailureResult(updateProvider)
+
+    assertThat(failure).hasMessageThat()
+      .contains("ProfileId ${PROFILE_ID_3?.internalId} does not match an existing Profile")
+  }
+
+  @Test
+  fun testUpdateProfile_updateProfileType_existingAdminProfile_checkUpdateSucceeded() {
+    setUpTestApplicationComponent()
+    profileTestHelper.addOnlyAdminProfile()
+
+    val updateProvider = profileManagementController.updateProfileType(
+      PROFILE_ID_0,
+      ProfileType.SUPERVISOR
+    )
+    monitorFactory.waitForNextSuccessfulResult(updateProvider)
+  }
+
+  @Test
+  fun testUpdateProfile_updateProfileType_existingNonAdminProfile_checkUpdateSucceeded() {
+    setUpTestApplicationComponent()
+    addNonAdminProfileAndWait(name = "Rajat", pin = "01234")
+
+    val updateProvider = profileManagementController.updateProfileType(
+      PROFILE_ID_0,
+      ProfileType.ADDITIONAL_LEARNER
+    )
+    monitorFactory.waitForNextSuccessfulResult(updateProvider)
```

Comment:
Should use ``ensureNextResultIsSuccess`` for asserting rather than 'wait' since it's more intentional.

Or, did you mean to actually verify the changed state? Probably relates to my comment in the other conversation thread.

Ditto for all other new tests in this file that aren't checking state.

Reply:
I have changed the test assertions a bit, so that I retrieve the profile and assert the value of its profileType field instead. I thinnk that is explicit as well.